flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/plan/rules/logical/LogicalCorrelateToJoinFromTemporalTableRule.scala

```diff
@@ -31,26 +32,65 @@ import org.apache.calcite.rel.logical.{LogicalCorrelate, LogicalFilter, LogicalS
   */
 class LogicalCorrelateToJoinFromTemporalTableRule
   extends RelOptRule(
-    operand(classOf[LogicalFilter],
-      operand(classOf[LogicalCorrelate], some(
-        operand(classOf[RelNode], any()),
-        operand(classOf[LogicalSnapshot], any())))),
+    operand(classOf[LogicalCorrelate],
+      operand(classOf[RelNode], any()),
+      operand(classOf[RelNode],
+        operand(classOf[RelNode], any()))),
     "LogicalCorrelateToJoinFromTemporalTableRule") {
 
+  override def matches(call: RelOptRuleCall): Boolean = {
+    val rel: RelNode = call.rel(2)
+    rel match {
+      case _: LogicalFilter =>
+        val input: RelNode = call.rel(3)
+        input.isInstanceOf[LogicalSnapshot]
```

Comment:
Can we just split this rule into 2 sub-rules, one is with top filter and one it not.